pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
@deads2k I agree with your concerns of keeping errors in their places. Will think about this separately, an error printer (or even a printer for *all* kinds of messages) in the command creation is one idea.